File: go/xss_vulnerable.go
package main

import (
	"fmt"
	"html/template"
	"net/http"
	_ "text/template"
)

// Matches: go-template-html (existing)
func vulnerableTemplateHTML(userInput string) template.HTML {
	return template.HTML(userInput)
}

// Matches: go-text-template-unescaped (pattern-regex matches import)
// The text/template import above triggers this rule

// Matches: go-fmt-fprintf-http
func vulnerableFprintf(w http.ResponseWriter, userInput string) {
	fmt.Fprintf(w, userInput)
}

// Safe: use html/template auto-escaping
func safeTemplate(w http.ResponseWriter, data interface{}) {
	tmpl := template.Must(template.New("page").Parse("<p>{{.}}</p>"))
	tmpl.Execute(w, data)
}

File: php/xss_vulnerable.php
<?php
// XSS Vulnerable Test Cases

// Matches: php-echo-unescaped
function vulnerable_echo($userInput) {
    echo $userInput;
}

// Matches: php-xss-print
function vulnerable_print($userInput) {
    print($userInput);
}

// Matches: php-blade-unescaped (pattern-regex)
// In Blade template: {!! $userInput !!}

// Safe: htmlspecialchars
function safe_echo($userInput) {
    echo htmlspecialchars($userInput, ENT_QUOTES, 'UTF-8');
}
